core/src/main/scala/shapeless/records.scala

```diff
@@ -152,4 +193,17 @@ class RecordMacros(val c: whitebox.Context) {
       case(elem, acc) => q""" $hconsValueTree(${promoteElem(elem)}, $acc) """
     }
   }
+
+  def mkParamsImpl[L <: HList](method: MethodSymbol, rec: Expr[L]): List[Tree] = {
+    def mkElem(keyTpe: Type, value: Tree): Tree =
+      q"implicitly[shapeless.ops.hlist.Selector[${rec.actualType}, ${mkFieldTpe(keyTpe, value.tpe)}]].apply($rec)"
```

Comment:
@beezee A general remark when writing macros: Non-interpolated terms and types in quasiquotes (like `implicitly` and `Selector`) should be fully qualified to avoid macro hygiene issues. So in this case you should use `_root_.scala.Predef.implicitly` and `_root_.shapeless.ops.hlist.Selector`.